python/ray/data/dataset.py

```diff
@@ -1706,7 +1728,9 @@ def min(
     ) -> U:
         """Compute minimum over entire dataset.
 
-        This is a blocking operation.
+        .. note::
+            This triggers execution of the lazy transformations performed on this
+            dataset, and will block until execution completes.
```

Comment:
nit: To avoid repeating ourselves, could we make a method annotation such as `@dataset.consumption_api` that injects this to the `__doc__`?

Reply:
(You can check out the PublicAPI annotation implementation for an example)

Reply:
Just tried this out locally, but I feel like the positioning of this `note` is pretty important, where ideally it would be before the examples or the args documentation.

I can write a modified version of `_append_doc()` that finds the start of the first non-summary block of text (i.e. looks for the first `"\n\n"`, if one exists, inserting one otherwise if an API is summary line only), and then insert this note there. How does that sound?